R CRAN warning in rray about xtensor

Hi xtensor team, I'm submitting a small patch release of rray to CRAN, and they responded saying that there was a significant warning in one of their test builds. It looks like this is the problem:

```c
/data/gannet/ripley/R/test-4.0/xtensor/include/xtensor/xstorage.hpp:1162:13: warning: array subscript 1 is outside array bounds of 'std::size_t [1]' {aka 'long unsigned int [1]'} [-Warray-bounds]
 1162 |             ++elt_ptr;
      |             ^~
In file included from positions.cpp:1:
/data/gannet/ripley/R/test-4.0/xtensor/include/xtensor/xsort.hpp:76:55: note: while referencing 'ax'
   76 |         get_permutations(std::size_t dim, std::size_t ax, layout_type layout)
      |                                           ~~~~~~~~~~~~^~
In file included from /data/gannet/ripley/R/test-4.0/xtensor/include/xtensor/xiterable.hpp:12,
                 from /data/gannet/ripley/R/test-4.0/xtensor/include/xtensor/xcontainer.hpp:21,
                 from /data/gannet/ripley/R/test-4.0/xtensor/include/xtensor-r/rarray.hpp:18,
                 from ../inst/include/rray.h:8,
                 from subset.cpp:1:
```

That was a slightly older version of xtensor compared to current master, but I think its pointing to this:
https://github.com/xtensor-stack/xtensor/blob/af5cc6c4891bde6946cb87120857fd72b6ae086a/include/xtensor/xstorage.hpp#L1168

This build uses GCC 10, so maybe you can reproduce if you test against that? The full log is here, with lots of other warnings that are deemed to not be significant:
https://www.stats.ox.ac.uk/pub/bdr/gcc10/rray.log

With a condensed version showing that specific "significant warning" here:
https://www.stats.ox.ac.uk/pub/bdr/gcc10/rray.out

They have only recently started testing with GCC 10, so that is why it hasn't shown up in the past.

I'm asking for rray to be pushed through anyways, I'm hopeful that they will allow it.

Maybe `-Warray-bounds` is relevant

--- a/include/xtensor/xstorage.hpp
+++ b/include/xtensor/xstorage.hpp
@@ -1164,10 +1164,10 @@ namespace xt
 
         // Update ref if element moved
         const T* elt_ptr = &elt;
-        if (it <= elt_ptr && elt_ptr < m_end)
-        {
-            ++elt_ptr;
-        }
+        bool cond = it <= elt_ptr && elt_ptr < m_end;
+        // More complicated than incrementing elt_ptr, but this avoids
+        // false positive array-bounds warning on GCC 10
+        const T* src_ptr = cond ? it + (elt_ptr - it) + std::ptrdiff_t(1) : elt_ptr;
         *it = *elt_ptr;
         return it;
     }

